pkg/reconciler/ingress/resources/names/names.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package names
+
+import (
+	"github.com/knative/pkg/kmeta"
+)
+
+// IngressVirtualService returns the name of the VirtualService child
+// resource for given Ingress that programs traffic for Ingress
+// Gateways.
+func IngressVirtualService(i kmeta.Accessor) string {
+	return i.GetName()
+}
+
+// MeshVirtualService returns the name of the VirtualService child
+// resource for given Ingress that programs traffic for Service
+// Mesh.
+func MeshVirtualService(i kmeta.Accessor) string {
+	return i.GetName() + "-mesh"
```

Comment:
This hack sort of work for ClusterIngress since we are naming it using UUID of the Route, and UUID lengths are ~30ish characters.

We may want to either use UUID again here (or may be use UUID only when the name is too long) or even use GenerateName to make sure this isn't too long.

It's ok to follow up with this in a different PR.

Reply:
(here I am assuming that for namespaced Ingress we will inherit the name of the Route onto the namespaced Ingress, which is consistent with many of our resources).

Reply:
I have been using the name of the route (i.e. service's name like "helloworld") instead of route-<uid>.   We really don't need the uid if we have namespace.   Is it your assumption?

Reply:
If the route name is long, like 64 characters, this VirtualService here will have a name of 69 characters and will be rejected.

We could follow this up in a separate PR.

Reply:
I will use UUID again to keep VS name within length limit.  It will be followed up in a separate PR.

Reply:
You should use the new method @vagababov is adding under `pkg/resources`.

Reply:
+1

Reply:
Now Using /pkg/resources/ChildName method to create names for new Ingress VirtualSerivces with -virtualservice/-virtualservice-mesh suffix